### DISCUSSION TITLE
is it possible to create elements between the link

### DISCUSSION THREAD

**Discussion by reshin-pulapaddy** [clientIO/joint#1968] [Q&A]
![MicrosoftTeams-image](https://user-images.githubusercontent.com/117301232/210201444-f16f1ff4-c4a0-4d9e-9a18-8f730502e389.png)
is it possible to create elements between the link?
how to put multiple elements in one row( I am using kitchen sink)

**Comment by kumilingus** [comment]
On general note:
1. You can create a series of elements with links between them
2. You can add link labels of arbitrary shapes *(label does not have to be a text with background)*
3. You can add elements next to the link and connect them to the link with another link (e.g. using the `linkAnchor` `connectionPerpendicular`).

It depends what exactly you try to achieve. Could you please elaborate more on the use-case?

**Reply by vyshak-pr** [reply]
@kumilingus 
![first](https://user-images.githubusercontent.com/120721238/210337308-8f053a1b-85a0-4bcc-8196-c552b3b175c9.png)

first I need to drag and drop a link(standard.link) from stencil paper to paper it is working fine.
![second](https://user-images.githubusercontent.com/120721238/210337378-be03f62f-4460-417f-9aac-62fd6ac8075b.jpg)
second When I click a custom button from the inspector part I need to add this element inside  the link (check the element from above image)
the element must be movable (like the label)
example image

![3](https://user-images.githubusercontent.com/120721238/210338720-8de85504-ff0e-4c2d-8e4e-ef6c4d576df1.png)

**Reply by kumilingus** [reply]
Ok, so why don't you use the link label for that? You can already add any number of labels in the Kitchen Sink application. You just need to change the appearance of the labels.

An example of custom labels is to be seen [here](https://www.jointjs.com/demos/links).

**Comment by vyshak-pr** [comment]
@kumilingus     

```js
var link = new joint.shapes.standard.Link({
            source: { x: 400, y: 200 },
            target: { x: 740, y: 200 },
            connector: { name: 'smooth' },
            attrs: {
                line: {
                    targetMarker: {
                        'd': 'M 0 -5 L -10 0 L 0 5 Z'
                    }
                }
            },
            labels: [{
                markup: [{
                    tagName: 'rect',
                    selector: 'labelBody'
                }, {
                    tagName: 'text',
                    selector: 'labelText'
                }],
                attrs: {
                    labelText: {
                        text: 'First',
                        fill: '#7c68fc',
                        fontFamily: 'sans-serif',
                        textAnchor: 'middle',
                        textVerticalAnchor: 'middle'
                    },
                    labelBody: {
                        ref: 'labelText',
                        x: 'calc(x - 5)',
                        y: 'calc(y - 5)',
                        width: 'calc(w + 10)',
                        height: 'calc(h + 10)',
                        stroke: '#7c68fc',
                        fill: 'white',
                        strokeWidth: 2,
                        rx: 5,
                        ry: 5
                    }
                },
                position: {
                    distance: 0.3,
                    args: {
                        keepGradient: true,
                        ensureLegibility: true,
                    }
                }
            }]
        });
```

I am using this link, but instead of a rect label I need a custom shape
example image:

![sample](https://user-images.githubusercontent.com/120721238/210362295-e977324e-c310-4090-bd8a-3c731cd3d055.png)
how can I create custom shape in label?

**Reply by kumilingus** [reply]
The link labels can be defined using arbitrary SVG tags (same as diagram elements and links).

Here's an example of custom labels: https://jsfiddle.net/kumilingus/rdo2g6z9/

**Reply by vyshak-pr** [reply]
Thank you @kumilingus
